### PR TITLE
add hidpi support.

### PR DESCRIPTION
**Pull request by lixiaoyan**

### CODE REVIEW

**Comment by mosch** [comment]
can you explain this change?

**Comment by lixiaoyan** [comment]
@mosch 
this pr added hidpi support for `<canvas>` element.
some information about hidpi canvas: <https://www.html5rocks.com/en/tutorials/canvas/hidpi/>.

**Comment by mosch** [comment]
looks like a great change, how can I test this?

**Comment by kuka** [comment]
@mosch pixels should be doubled as I understand. This PR would be on point, any help needed?

File: index.js [hunk 1]
```diff
@@ -49,6 +49,9 @@ const draggableEvents = {
 }
 const deviceEvents = isTouchDevice ? draggableEvents.touch : draggableEvents.desktop
 
+const pixelRatio = typeof window !== 'undefined' && typeof window.devicePixelRatio !== 'undefined'
+  ? window.devicePixelRatio : 1
+
```
Comment (hu9o) [inline index.js]: this could be summarized as:
`const pixelRatio = typeof window !== 'undefined' && window.devicePixelRatio || 1;`

Comment (hu9o) [inline index.js]: Also, be aware that `window.devicePixelRatio` is affected by the _zoom_ feature of browsers, meaning **its value can change at any time**. It's probably not a big deal though.

Comment (lixiaoyan) [inline index.js]: Should we check the `devicePixelRatio` each time when rendering?

**Comment by lixiaoyan** [comment]
🎉 
Can you make a new release for this? @mosch

**Pull request merged**